### PR TITLE
Logs export OTLP

### PR DESCRIPTION
**Pull request by breedx-splk**
Now that logs are being batched, we can export them to the collector via OTLP.

### CODE REVIEW

File: profiler/src/main/java/com/splunk/opentelemetry/logs/BatchingLogsProcessor.java [hunk 1]
```diff
@@ -85,7 +85,8 @@ public void log(LogEntry log) {
   private void asyncDoExport() {
     List<LogEntry> batchCopy = copyClearBatch();
     if (batchCopy != null) {
-      executorService.submit(() -> exporter.export(batchCopy));
+      Runnable task = HelpfulExecutors.logUncaught(() -> exporter.export(batchCopy));
```
Comment (jkwatson) [inline profiler/src/main/java/com/splunk/opentelemetry/logs/BatchingLogsProcessor.java:88]: this method seems like an odd one to be hanging off the `HelpfulExecutors` class. also, the name would imply to me that I was getting a helpful executor back, when all I get is a runnable.

File: profiler/src/main/java/com/splunk/opentelemetry/logs/InstrumentationLibraryLogsAdapter.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    InstrumentationLibrary library =
+        InstrumentationLibrary.newBuilder()
+            .setName(instrumentationName)
+            .setVersion(instrumentationVersion)
+            .build();
+
+    List<LogRecord> logs = logEntries.stream().map(logEntryAdapter).collect(Collectors.toList());
+
+    return InstrumentationLibraryLogs.newBuilder()
+        .setInstrumentationLibrary(library)
```
Comment (jkwatson) [inline profiler/src/main/java/com/splunk/opentelemetry/logs/InstrumentationLibraryLogsAdapter.java:55]: Should we also populate the otel `schemaUrl` from the SemanticConventions we're relying on here?

File: profiler/src/main/java/com/splunk/opentelemetry/logs/ResourceLogsAdapter.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  @Override
+  public ResourceLogs apply(List<LogEntry> logEntry) {
+    InstrumentationLibraryLogs instrumentationLibraryLogs = logsAdapter.apply(logEntry);
+    io.opentelemetry.proto.resource.v1.Resource protoResource =
+        io.opentelemetry.proto.resource.v1.Resource.newBuilder()
+            .addAllAttributes(resourceAttributes)
+            .build();
+    return ResourceLogs.newBuilder()
```
Comment (jkwatson) [inline profiler/src/main/java/com/splunk/opentelemetry/logs/ResourceLogsAdapter.java:56]: I think we should be setting the `schemaUrl` here, as well.

Comment (jkwatson) [reply profiler/src/main/java/com/splunk/opentelemetry/logs/ResourceLogsAdapter.java:56]: (note this one should come from the `ResourceAttributes.SCHEMA_URL` constant.

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/context/SpanContextualizer.java [hunk 4]
```diff
@@ -53,27 +54,27 @@ public void updateContext(RecordedEvent event) {
   }
 
   /** Parses thread info from the raw stack string and links it to a span (if available). */
-  public StackToSpanLinkage link(Instant time, String rawStack) {
+  public StackToSpanLinkage link(Instant time, String rawStack, RecordedEvent event) {
```
Comment (jkwatson) [inline profiler/src/main/java/com/splunk/opentelemetry/profiler/context/SpanContextualizer.java:57]: these changes seem relatively unrelated to the export pipeline. Any reason this couldn't be split out into a separate PR?

File: profiler/src/main/java/com/splunk/opentelemetry/logs/ResourceLogsAdapter.java (hunk 3 above)
Comment (jkwatson) [reply profiler/src/main/java/com/splunk/opentelemetry/logs/ResourceLogsAdapter.java:56]: hmm. looks like we're only using semconv for testing right now?

File: profiler/src/main/java/com/splunk/opentelemetry/logs/BatchingLogsProcessor.java (hunk 1 above)
Comment (breedx-splk) [reply profiler/src/main/java/com/splunk/opentelemetry/logs/BatchingLogsProcessor.java:88]: Yeah, it made more sense in there before I fixed a bug.  🤣 
I can move it out.

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/LogEntryCreator.java [hunk 5]
```diff
@@ -32,21 +35,36 @@
...
+    return builder.build();
   }
 
-  private Attributes buildAttributes(StackToSpanLinkage stackToSpanLinkage) {
-    String sourceEvent = stackToSpanLinkage.getSourceEventName();
+  private Attributes buildAttributes(StackToSpanLinkage linkedStack) {
+    String sourceEvent = linkedStack.getSourceEventName();
     long eventPeriodMs = -999; // tbd
+
+    // Note: It is currently believed that he span id and trace id on the LogRecord itself
```
Comment (mateuszrzeszutek) [inline profiler/src/main/java/com/splunk/opentelemetry/profiler/LogEntryCreator.java:54]:
```suggestion
    // Note: It is currently believed that the span id and trace id on the LogRecord itself
```

File: profiler/src/main/java/com/splunk/opentelemetry/logs/CommonAdapter.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+import io.opentelemetry.proto.common.v1.InstrumentationLibrary;
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import java.util.List;
+
+/** Copied from otel internals, because it's so locked down we can't use it here. */
+final class CommonAdapter {
+  @SuppressWarnings("unchecked")
+  public static KeyValue toProtoAttribute(AttributeKey<?> key, Object value) {
+    KeyValue.Builder builder = KeyValue.newBuilder().setKey(key.getKey());
```
Comment (mateuszrzeszutek) [inline profiler/src/main/java/com/splunk/opentelemetry/logs/CommonAdapter.java:31]: `builder` is not used in the switch statement at all; how about moving it just before the last `return`?

**Review by mateuszrzeszutek — APPROVED** [review]

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/context/SpanContextualizer.java (hunk 4 above)
Comment (breedx-splk) [reply profiler/src/main/java/com/splunk/opentelemetry/profiler/context/SpanContextualizer.java:57]: It would be a lot of work at this point because the PRs were already stacking up...it was a decent amount of work to get to this stage already.  🙏🏻

File: profiler/src/main/java/com/splunk/opentelemetry/logs/CommonAdapter.java (hunk 6 above)
Comment (breedx-splk) [reply profiler/src/main/java/com/splunk/opentelemetry/logs/CommonAdapter.java:31]: Sure. I copied it outright and didn't add any tests nor did I attempt any cleanup.  :)

File: profiler/src/main/java/com/splunk/opentelemetry/logs/ResourceLogsAdapter.java (hunk 3 above)
Comment (breedx-splk) [reply profiler/src/main/java/com/splunk/opentelemetry/logs/ResourceLogsAdapter.java:57]: Yeah, looks like it.

**Comment by iNikem** [comment]
@breedx-splk Why did you put this into Splunk repo and not into Otel Java SDK?

**Comment by jkwatson** [comment]
> @breedx-splk Why did you put this into Splunk repo and not into Otel Java SDK?

We'll need to release profiling as non-beta long before logs will be declared stable in the SDK. And, I think we didn't want to wait a month for the Java SDK to be updated.

The plan is to upstream this when the SDK is ready for it and ready to make logs stable.
